### DISCUSSION TITLE
How do you safelist a class name with the important !prefix?

### DISCUSSION THREAD

**Discussion by hillin** [tailwindlabs/tailwindcss#8249] [Help] [answered]
I have a series of class names that are generated with javascript. I've already listed them in the `safelist` section. Now I'd like to make these classes important with the `!` prefix, but it seems I can't simply place a `!` in the generated class name, neither does prefixing the safelisted class names work.

**Comment by abdulhadi-lababidi** [comment]
any updates on this?

**Comment by wongjn** [comment]
What version of Tailwind are you running? [Just tried on the playground](https://play.tailwindcss.com/PTgY8aUpR4?file=config) and `!bg-black` works.

**Reply by marchjorth** [reply, marked as answer]
Works in v3.1.8 - doesn't work in v2 😶

**Reply by iandoesallthethings** [reply]
Have you or anyone gotten it to work with pattern matching? Safelisting works with regexes like `/(fill|border|bg|text)-(.*)-(\d{1}0{1,2})/`, but `/!?(fill|border|bg|text)-(.*)-(\d{1}0{1,2})/` won't safelist important classes even though the [regex works](regexr.com/713ri) according to regexr.
